RecoLocalCalo/HcalRecAlgos/interface/HcalSimpleRecAlgo.h

```diff
@@ -111,8 +108,13 @@ class HcalSimpleRecAlgo {
 
   std::auto_ptr<PulseShapeFitOOTPileupCorrection> psFitOOTpuCorr_;
   
+  std::auto_ptr<PedestalSub> pedSubFxn_;
+
   // S.Brandt Feb19 : Add a pointer to the HLT algo
   std::auto_ptr<HcalDeterministicFit> hltOOTpuCorr_;
+
+  // Speed up the code a little bit by not allocating a vector every time
+  mutable std::vector<double> vectorBuffer_;
```

Comment:
The allocations internal to "Method 2" and "Method 3" classes are, of course, irrelevant to this discussion. Bug the authors of the corresponding classes if you are so inclined.

The change with a mutable, of course, does not open any possibility to pass information between hits. The vector is cleared any time it is used.

Any code is "fragile" to some degree. I believe, the advantages outweigh the drawbacks in this particular case. The "mutable" keyword is included into the language precisely for the uses similar to this.

The cost reasons are obvious: we avoid allocating the contents of the vector on the heap for every hcal rechit.